lib/fog/aws/kinesis.rb

```diff
@@ -0,0 +1,157 @@
+module Fog
+  module AWS
+    class Kinesis < Fog::Service
+      extend Fog::AWS::CredentialFetcher::ServiceMethods
+
+      class LimitExceeded < Fog::Errors::Error; end
+      class ResourceNotFound < Fog::Errors::Error; end
+      class ExpiredIterator < Fog::Errors::Error; end
+      class InvalidArgument < Fog::Errors::Error; end
+      class ProvisionedThroughputExceeded < Fog::Errors::Error; end
+
+      requires :aws_access_key_id, :aws_secret_access_key
+      recognizes :region, :host, :path, :port, :scheme, :persistent, :use_iam_profile, :aws_session_token, :aws_credentials_expire_at, :instrumentor, :instrumentor_name
+
+      request_path 'fog/aws/requests/kinesis'
+
+      request :list_streams
+      request :describe_stream
+      request :create_stream
+      request :delete_stream
+      request :get_shard_iterator
+      request :put_records
+      request :get_records
+
+      class Real
+        include Fog::AWS::CredentialFetcher::ConnectionMethods
+
+        def initialize(options={})
+          @use_iam_profile = options[:use_iam_profile]
+
+          @connection_options = options[:connection_options] || {}
+
+          @instrumentor           = options[:instrumentor]
+          @instrumentor_name      = options[:instrumentor_name] || 'fog.aws.kinesis'
+
+          options[:region] ||= 'us-east-1'
+          @region     = options[:region]
+          @host       = options[:host] || "kinesis.#{options[:region]}.amazonaws.com"
+          @path       = options[:path]        || '/'
+          @persistent = options[:persistent]  || false
+          @port       = options[:port]        || 443
+          @scheme     = options[:scheme]      || 'https'
+          @connection = Fog::XML::Connection.new("#{@scheme}://#{@host}:#{@port}#{@path}", @persistent, @connection_options)
+          @version    = "20131202"
+
+          setup_credentials(options)
+        end
+
+        private
+
+        def setup_credentials(options)
+          @aws_access_key_id      = options[:aws_access_key_id]
+          @aws_secret_access_key  = options[:aws_secret_access_key]
+          @aws_session_token      = options[:aws_session_token]
+          @aws_credentials_expire_at = options[:aws_credentials_expire_at]
+
+          @signer = Fog::AWS::SignatureV4.new( @aws_access_key_id, @aws_secret_access_key, @region, 'kinesis')
```

Comment:
I know performance had been a concern around some of this. We might be able to optimize the v4 signer also, I think it was a bit of a slowdown compared to v3, and I'm not sure anyone has looked at it with an eye toward performance in particular.